task3.cpp

```diff
@@ -0,0 +1,225 @@
+		strCopy(this->title, title);
+		this->price = price;
+		this->isAvailable = isAvailable;
+	}
+
+	const char* getTitle() const {
+		return this->title;
+	}
+	
+	const int getPrice() const {
+		return this->price;
+	}
+
+	const bool getAvailabilty() const {
+		return this->isAvailable;
+	}
+
+	void setTitle(const char* title) {
+		strCopy(this->title, title);
+	}
+
+	void setPrice(const int price) {
+		this->price = price;
+	}
+
+	void setAvailability(const bool isAvailable) {
+		this->isAvailable=isAvailable;
+	}
+
+	bool isFree() const {
+		if (this->price == 0) {
+			return true;
+		}
+		return false;
+	}
+
+	void print() const {
+		std::cout << "Game title: " << this->title<<std::endl;
+		std::cout << "Game price: " << this->price << std::endl;
+		std::cout << std::boolalpha << "Is the game available" << this->isAvailable << std::endl;
+	}
+
+
+	void writeGameToFile(std::ofstream& out) const {
+		if (!out.is_open()) {
+			return;
+		}
+		out << this->getTitle() << this->getPrice() << this->getAvailabilty();
+	}
+
+	void readGameFromFile(std::ifstream& in) {
+		if (!in.is_open()) {
+			return;
+		}
+
+		in >> this->title >> this->price >> this->isAvailable;
+	}
+
+	~Game() {};
+};
+
+class GamePlatform {
+private:
+	Game* Games;
+	int size = 0;
+public:
+	GamePlatform() {
+		this->Games = nullptr;
+		this->size = 0;
+	};
+
+	GamePlatform(const char* title,const int price,const  bool isAvailable) {
+		if (this->size > MAX_GAMES) {
+			std::cout << "You hava reached the max number of games" << std::endl;
+			return;
+		}
+		this->Games[this->size].setTitle(title);
+		this->Games[this->size].setPrice(price);
+		this->Games[this->size].setAvailability(isAvailable);
+		this->size++;
+	}
+
+	void printGame(const int index) const {
+		this->Games[index].print();
+	}
+
+	void printAllGame() const {
+		for (int i = 0; i < this->size; i++) {
+			this->Games[i].print();
+		}
+	}
+		
+	void cheapestGame() const {
+		int min = MAX;
+		for (int i = 0; i < this->size; i++) {
+			if (this->Games[i].getPrice() < min) {
+				min = this->Games[size].getPrice();
+			}
+		}
+		for(int i=0;i<this->size;i++){
+			if (this->Games[i].getPrice() == min) {
+				this->Games[i].print();
+			}
+		}
+	}
+
+	void mostExpensiveGame() const {
+		int max = MIN;
+		for (int i = 0; i < this->size; i++) {
+			if (this->Games[i].getPrice() > max) {
+				max = this->Games[size].getPrice();
+			}
+		}
+		for (int i = 0; i < this->size; i++) {
+			if (this->Games[i].getPrice() == max) {
+				this->Games[i].print();
+			}
+		}
+	}
+
+	void printAllFreeGame() const {
+		for (int i = 0; i < this->size; i++) {
+			if (this->Games[i].getPrice() == 0) {
+				this->Games[i].print();
+			}
+		}
+	}
+
+	void removeGame(const char* titleToRemove) {
+		GamePlatform Games2;
+		for (int i = 0; i < this->size; i++) {
+			if (strcmp(this->Games[i].getTitle(), titleToRemove) != 0) {
+				Games2.Games[i] = this->Games[i];
+			}
+		}
+		this->Games = Games2.Games;
+		this->size--;
```

Comment:
it's wrong
you have to find the index of the game with that title if there is a  such game
then place the last game at that index and just decrement the size.
since the Game class does not have dynamic memory, you do not have to delete anything